apps/gitness/src/pages-v2/pull-request/__tests__/pull-request-utils.test.ts

```diff
@@ -0,0 +1,253 @@
+import { describe, expect, it } from 'vitest'
+
+import {
+  EnumCheckStatus,
+  EnumPullReqReviewDecision,
+  TypesCodeOwnerEvaluationEntry,
+  TypesOwnerEvaluation,
+  TypesPrincipalInfo
+} from '@harnessio/code-service-client'
+
+import { PullReqReviewDecision, TypeCheckData } from '../../../pages/pull-request/types/types'
+import {
+  capitalizeFirstLetter,
+  changedFileId,
+  checkIfOutdatedSha,
+  determineStatusMessage,
+  extractInfoForCodeOwnerContent,
+  extractInfoFromRuleViolationArr,
+  extractSpecificViolations,
+  findChangeReqDecisions,
+  findWaitingDecisions,
+  generateAlphaNumericHash,
+  generateStatusSummary,
+  getErrorMessage,
+  normalizeGitFilePath,
+  processReviewDecision
+} from '../pull-request-utils'
+
+// Mock data
+const mockChecks = [
+  { check: { status: 'success' as EnumCheckStatus }, required: true, bypassable: false },
+  { check: { status: 'failure' as EnumCheckStatus }, required: false, bypassable: false }
+]
+
+const mockViolationsData = {
+  data: {
+    rule_violations: [
+      { violations: [{ code: 'rule1', description: 'Violation 1' }] },
+      { violations: [{ code: 'rule2', description: 'Violation 2' }] }
+    ]
+  }
+}
+
+const mockReviewDecision = 'approved' as EnumPullReqReviewDecision
```

Comment:
And `EnumPullReqReviewDecision` isn't an enum either.

Reply:
done